web/beacon-app/src/features/topics/components/RevokeAPIKeyModal.tsx

```diff
@@ -13,14 +14,47 @@ type RevokeAPIKeyModalProps = {
   onClose: () => void;
 };
 
+// TODO: This component needs to be refactored
+
 const RevokeAPIKeyModal = ({ onOpen, onClose }: RevokeAPIKeyModalProps) => {
   const { opened, key } = onOpen;
+  const { deleteApiKey, wasKeyDeleted, hasKeyDeletedFailed, reset, isDeletingKey, error } =
+    useDeleteAPIKey(key?.id || '');
+
   const [isChecked, setIsChecked] = useState(false);
 
+  // eslint-disable-next-line react-hooks/exhaustive-deps
   const handleCheckboxChange = () => {
     setIsChecked(!isChecked);
   };
 
+  const deleteAPIKeyHandler = () => {
+    deleteApiKey();
+  };
+
+  useEffect(() => {
+    if (hasKeyDeletedFailed) {
+      reset();
+      handleCheckboxChange();
+      onClose();
+
+      toast.error(
+        error?.response?.data?.error ||
+          t`Sorry, but we had a problem while trying to revoke the API key. Please try again. If the issue persists, contact our support team for assistance.`
```

Comment:
```suggestion
          t`Sorry, we were unable to revoke the API key. Please try again. If the issue persists, contact our support team for assistance.`
```

Reply:
A suggestion for making the message a little shorter since it's on the screen for a few seconds.